oak_sign/src/main.rs

```diff
@@ -124,96 +93,31 @@ fn main() -> anyhow::Result<()> {
-            let key_pair = signature::Ed25519KeyPair::from_pkcs8(private_key_bytes.as_ref())
-                .expect("Couldn't parse generated key pair");
-            let public_key_bytes = key_pair.public_key();
-
-            // Encode key pair in PEM format.
-            let private_key_pem = create_pem(PRIVATE_KEY_TAG, private_key_bytes.as_ref());
-            let encoded_private_key = encode(&private_key_pem);
-
-            let public_key_pem = create_pem(PUBLIC_KEY_TAG, public_key_bytes.as_ref());
-            let encoded_public_key = encode(&public_key_pem);
-
-            // Write key files.
-            write(&opt.private_key, encoded_private_key)
-                .with_context(|| format!("Couldn't write private key file {}", &opt.private_key))?;
-            write(&opt.public_key, encoded_public_key)
-                .with_context(|| format!("Couldn't write public key file {}", &opt.public_key))?;
+            let key_pair = KeyPairStruct::generate()?;
+            write_pem_file(&opt.private_key, PRIVATE_KEY_TAG, &key_pair.private_key)?;
+            write_pem_file(&opt.public_key, PUBLIC_KEY_TAG, &key_pair.public_key)?;
             info!("Key pair generated successfully");
         }
         Command::Sign(ref opt) => {
-            // Read input files.
-            let private_key_file = read(&opt.private_key)
-                .with_context(|| format!("Couldn't read private key file {}", &opt.private_key))?;
-            let input_file_bytes = read(&opt.input_file)
-                .with_context(|| format!("Couldn't read input file {}", &opt.input_file))?;
-
-            // Sign input file.
-            let private_key_pem =
-                parse(private_key_file).context("Couldn't parse PEM encoded private key file")?;
-            let key_pair =
-                signature::Ed25519KeyPair::from_pkcs8(&private_key_pem.contents.as_ref())
-                    .expect("Couldn't parse PKCS8 encoded private key");
-            let public_key_bytes = key_pair.public_key();
-            let signature_bytes = key_pair.sign(&input_file_bytes);
-            let hash_bytes = get_sha256(&input_file_bytes);
-
-            // Encode signature in PEM format.
-            let public_key_pem = create_pem(PUBLIC_KEY_TAG, public_key_bytes.as_ref());
-            let signature_pem = create_pem(SIGNATURE_TAG, signature_bytes.as_ref());
-            let hash_pem = create_pem(HASH_TAG, hash_bytes.as_ref());
-            let encoded_signature = encode_many(&[public_key_pem, signature_pem, hash_pem]);
-
-            // Write signature file.
-            write(&opt.signature, &encoded_signature)
-                .with_context(|| format!("Couldn't write signature file {}", &opt.signature))?;
+            let private_key = read_pem_file(&opt.private_key)?;
+            let input = match (&opt.input_file, &opt.input_string) {
+                (Some(_), Some(_)) => Err(anyhow!(
+                    "Both `input_file` and `input_string` are specified"
```

Comment:
I think it would be more useful if the message said what should be done, e.g. "Exactly one between `input_file` and `input_string` must be specified" (and below).